src/lib/io/sol-iio.c

```diff
@@ -366,6 +367,24 @@ set_buffer_enabled(struct sol_iio_device *device, bool enabled)
     return true;
 }
 
+static bool
+set_channel_enabled(struct sol_iio_device *device, const char *channel_name, bool enabled)
+{
+    char path[PATH_MAX];
+    bool r;
+
+    r = craft_filename_path(path, sizeof(path), CHANNEL_ENABLE_PATH, device->device_id, channel_name);
+    if (!r)
+        return false;
+
+    if (check_file_existence(path)) {
```

Comment:
@edersondisouza 
I need to craft path for sol_util_write_file(). Should I change check_file_existence to !stat(path, &st)  ?